src/intrprtr.c

```diff
@@ -4583,19 +4583,28 @@ void            IntrInfoBegin( void )
 }
 
 static Obj InfoDecision;
-static Obj InfoDecisionFast;
 static Obj IsInfoClassListRep;
 
 Obj InfoCheckLevel(Obj selectors, Obj level)
 {
-    if (IS_POS_INTOBJ(level) &&
-        CALL_1ARGS(IsInfoClassListRep, selectors) == True) {
+    // Fast-pass the most common failing case
+    // The fast-pass never returns True because the GAP function
+    // InfoDecision sets a number of variables as well as returning True
+    if (CALL_1ARGS(IsInfoClassListRep, selectors) == True) {
+#if defined(HPCGAP)
+        Obj index = ElmAList(selectors, 1);
+#else
         Obj index = ELM_PLIST(selectors, 1);
-        return CALL_3ARGS(InfoDecisionFast, index, selectors, level);
-    }
-    else {
-        return CALL_2ARGS(InfoDecision, selectors, level);
+#endif
```

Comment:
Because `ELM_LIST` doesn't work on positional objects, or atomic positional objects.

If we make it work, then given a positional object `X[3]` starts working at the GAP level, rather than the `X![3]` we expect people to use.

However, some kind of unified access would be nice, as this code is fairly horrible and I wouldn't want to start writing this regularily.

Reply:
OK, that makes sense.